l10n_it_delivery_note/models/account_invoice.py

```diff
@@ -123,22 +123,38 @@ def update_delivery_note_lines(self):
                     )
                 )
             else:
-                for line in invoice.invoice_line_ids:
-                    sequence = line.sequence - 1
-                    delivery_note_line = invoice.mapped(
-                        "delivery_note_ids.line_ids"
-                    ) & line.mapped("sale_line_ids.delivery_note_line_ids")
-                    for delivery_note_id in delivery_note_line.filtered(
+                sequence = 1
+                done_invoice_lines = self.env['account.invoice.line']
+                for dn in invoice.mapped(
+                        'invoice_line_ids.sale_line_ids.delivery_note_line_ids.'
+                        'delivery_note_id').sorted(key='name'):
+                    dn_invoice_lines = invoice.invoice_line_ids.filtered(
+                        lambda x: x not in done_invoice_lines and dn in x.mapped(
+                            'sale_line_ids.delivery_note_line_ids.delivery_note_id'
+                        )
+                        # fixme test invoice from 2 sale lines
+                    )
+                    done_invoice_lines |= dn_invoice_lines
+                    for note_line in dn.line_ids.filtered(
                         lambda l: l.invoice_status == DOMAIN_INVOICE_STATUSES[2]
-                    ).mapped("delivery_note_id"):
-                        line.delivery_note_id = delivery_note_id.id
-                        new_lines.append(
-                            (
-                                0,
-                                False,
-                                self._prepare_note_dn_value(sequence, delivery_note_id),
-                            )
+                    ):
+                        for invoice_line in dn_invoice_lines:
+                            if note_line in invoice_line.sale_line_ids.\
+                                    delivery_note_line_ids:
+                                invoice_line.delivery_note_id = note_line.\
+                                    delivery_note_id.id
+                    new_lines.append(
+                        (
+                            0,
+                            False,
+                            self._prepare_note_dn_value(
+                                sequence, dn
+                            ),
                         )
+                    )
+                    for invoice_line in dn_invoice_lines:
+                        sequence += 1
+                        invoice_line.sequence = sequence
```

Comment:
Questa parte corregge l'inserimento della nota del DDT in fattura, riportandolo solo una volta quando serve, invece di n volte. Inoltre li riporta in ordine di numero del DDT.